### PR TITLE
Develop blog project

### PR DESCRIPTION
**Pull request by Whitney-Rene**
On this branch, I create:

- my server.js and its routes
- .env file, utils.js and db folder
- components for viewing a preview of all blogpost, creating a new blogpost and I am currently working on the component that allows the user to click and see all the information related to a single blogpost

On the next iteration of this branch, I hope:
- complete the project (viewindblogpost comp)
- add lovely css

### CODE REVIEW

**Comment by destineevaldivia** [comment]
The code formatting and style look really clean! Your comments are informative and concise, and I really like that you include comments to explain your intentions for next steps. It's helpful for getting a full mental picture of your project. Keep up the great work!

**Comment by edDoganFreenome** [comment]
Looks great overall, a couple of style and preference issues as we talked about in our session. Great job!

File: client/src/components/CreateBlogPost.jsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    const handleSubmit = async (event) => {
+
+        event.preventDefault();
+
+        const newBlogPost = { blog_title: blogTitle.current?.value, blog_content: blogContent.current?.value, blog_picture: blogPicture.current?.value}
+        
+        try {
+            await handlePostRequest(newBlogPost);
+
+            console.log(newBlogPost)
```
Comment (plee-fn) [inline client/src/components/CreateBlogPost.jsx:41]: consider cleaning up console.logs

File: client/src/App.css [hunk 2]
```diff
@@ -40,3 +40,5 @@
 .read-the-docs {
   color: #888;
 }
+
+/* need to add css, background color, shading behind photos, form, buttons */
```
Comment (plee-fn) [inline client/src/App.css:44]: best practice is to leave an empty line as the very last line of the file. Consider adding an empty line after this line

File: client/src/app.test.jsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import App from './App';
+import { expect, test } from 'vitest';
+import { render, waitFor } from '@testing-library/react';
+
+test ('note renders', async () => {
+    const { getByText } = render(<App />);
+    await waitFor(() => getByText('You can do hard things, Whitney-Rene'));
+    expect(getByText('You can do hard things, Whitney-Rene')).toBeDefined();
+});
```
Comment (plee-fn) [inline client/src/app.test.jsx:9]: best practice is to leave an empty line as the very last line of the file. Consider adding an empty line after this line

File: client/src/components/CreateBlogPost.jsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                <button  className='createButton' type='submit'>Create New Blog Post</button>
+            </div>
+
+        </form>
+        </>
+    )
+
+}
+
+export default CreateBlogPost;
```
Comment (plee-fn) [inline client/src/components/CreateBlogPost.jsx:78]: best practice is to leave an empty line as the very last line of the file. Consider adding an empty line after this line

File: client/src/components/ViewBlogsPreview.jsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+                </div>
+
+            ))}
+            </div>
+
+        </>
+    )
+};
+
+export default ViewBlogsPreview;
```
Comment (plee-fn) [inline client/src/components/ViewBlogsPreview.jsx:51]: best practice is to leave an empty line as the very last line of the file. Consider adding an empty line after this line

File: client/src/components/ViewIndividualBlogModal.jsx [hunk 6]
```diff
@@ -0,0 +1,10 @@
+
+
+function ViewIndividualBlogModal () {
+
+// user will click on icon and see modal with all info from blogpost table in postgres
+//will I need to call an instace of this component within another component?
+
+}
+
+export default ViewIndividualBlogModal
```
Comment (plee-fn) [inline client/src/components/ViewIndividualBlogModal.jsx:10]: best practice is to leave an empty line as the very last line of the file. Consider adding an empty line after this line

File: client/src/utils.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import moment from 'moment';
+
+export const apiUrl = 'http://localhost:1012';
+
+export function formatTime (rawDate) {
+    const dateObject = moment(rawDate);
+    const formattedDate = dateObject.format('MMMM D, YYYY');
+    return formattedDate;
+}
```
Comment (plee-fn) [inline client/src/utils.js:9]: best practice is to leave an empty line as the very last line of the file. Consider adding an empty line after this line

File: client/vitest.config.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+
+// https://vitejs.dev/config/
+export default defineConfig({
+    plugins: [react()],
+    test: {
+        framework: 'vitest',
+        globals: true,
+        environment: 'jsdom'
+    }
+})
```
Comment (plee-fn) [inline client/vitest.config.js:14]: best practice is to leave an empty line as the very last line of the file. Consider adding an empty line after this line

File: server/db/db-connection.js [hunk 9]
```diff
@@ -0,0 +1,9 @@
+const { Pool } = require('pg');
+
+console.log(process.env.DB_URI)
+
+const db = new Pool ({
+    connectionString: process.env.DB_URI,
+});
+
+module.exports = db;
```
Comment (plee-fn) [inline server/db/db-connection.js:9]: best practice is to leave an empty line as the very last line of the file. Consider adding an empty line after this line

File: server/db/db.sql [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+--helpful statements
+UPDATE table_name
+SET column_name = new_value
+WHERE some_column = some_value;
+
+--inner join
+select blogpost.blog_picture, blogpost.blog_title, blogpost.blog_publishdate, authors.author_name FROM blogpost INNER JOIN authors on blogpost.author_id = authors.author_id;
+
+--quit postgres
+\q
```
Comment (plee-fn) [inline server/db/db.sql:53]: best practice is to leave an empty line as the very last line of the file. Consider adding an empty line after this line

File: server/server.js [hunk 11]
```diff
@@ -1,8 +1,58 @@
...
+        );
+        res.status(201).json(result.rows[0]);
+    } catch (error) {
+        return res.status(400).json({error})
+    }
+})
+
+app.listen(PORT, () => {
+    console.log(`Estoy escuchando en ${PORT}`)
+})
```
Comment (plee-fn) [inline server/server.js:58]: best practice is to leave an empty line as the very last line of the file. Consider adding an empty line after this line

File: client/src/App.jsx [hunk 12]
```diff
@@ -1,10 +1,38 @@
...
+
+    <h4>Whitney-Rene's Spain Blog</h4>
+    
+    <ViewBlogsPreview />
+
+    <ViewIndividualBlogModal />
+
+    <CreateBlogPost />
+
+    <p>{note}</p>
```
Comment (plee-fn) [inline client/src/App.jsx:34]: needs to be indented once as it's within the React.Fragment

**Comment by plee-fn** [comment]
Overall, the code so far is very clean aside from small minor things.

File: client/src/utils.js [hunk 13]
```diff
@@ -0,0 +1,9 @@
+import moment from 'moment';
+
+export const apiUrl = 'http://localhost:1012';
```
Comment (plee-fn) [inline client/src/utils.js:3]: imo `apiUrl` isn't a util... I consider utils as functions... I usually create another file called `const.js`, which will hold all my `export const` stuff.

This really depends on each person.

File: client/src/components/CreateBlogPost.jsx (hunk 1 above)
Comment (Whitney-Rene) [reply client/src/components/CreateBlogPost.jsx:41]: Thank you Philip!

File: client/src/App.css (hunk 2 above)
Comment (Whitney-Rene) [reply client/src/App.css:44]: Ah, yes, thank you for the reminder!

File: client/src/app.test.jsx (hunk 3 above)
Comment (Whitney-Rene) [reply client/src/app.test.jsx:9]: Done!

File: client/src/components/CreateBlogPost.jsx (hunk 4 above)
Comment (Whitney-Rene) [reply client/src/components/CreateBlogPost.jsx:78]: Thank you!

File: client/src/components/ViewIndividualBlogModal.jsx (hunk 6 above)
Comment (Whitney-Rene) [reply client/src/components/ViewIndividualBlogModal.jsx:10]: Thank you!

File: client/src/utils.js (hunk 7 above)
Comment (Whitney-Rene) [reply client/src/utils.js:9]: Thank you!

File: server/db/db-connection.js (hunk 9 above)
Comment (Whitney-Rene) [reply server/db/db-connection.js:9]: Thank you!

File: server/server.js (hunk 11 above)
Comment (Whitney-Rene) [reply server/server.js:58]: Thank you!

File: client/src/App.jsx (hunk 12 above)
Comment (Whitney-Rene) [reply client/src/App.jsx:34]: Perfect!  I needed that reminder.

**Comment by Whitney-Rene** [comment]
Thank you all for your feedback!

**Pull request merged**